drivers/display/display_smartbond.c

```diff
@@ -0,0 +1,634 @@
+/*
+ * Copyright (c) 2023 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT  renesas_smartbond_display
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/smartbond_clock_control.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/dma.h>
+#include <da1469x_lcdc.h>
+#include <DA1469xAB.h>
+#include <da1469x_pd.h>
+#include <zephyr/linker/devicetree_regions.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(smartbond_display, CONFIG_DISPLAY_LOG_LEVEL);
+
+#define SMARTBOND_IRQN		DT_INST_IRQN(0)
+#define SMARTBOND_IRQ_PRIO  DT_INST_IRQ(0, priority)
+
+#define LCDC_SMARTBOND_CLK_DIV(_freq)                           \
+	((32000000U % (_freq)) ? (96000000U / (_freq)) : (32000000U / (_freq)))
+
+#define LCDC_SMARTBOND_IS_PLL_REQUIRED \
+	!!(32000000U % DT_PROP(DT_INST_CHILD(0, display_timings), clock_frequency))
+
+#define DISPLAY_SMARTBOND_IS_DMA_PREFETCH_ENABLED \
+	DT_INST_ENUM_IDX_OR(0, dma_prefetch, 0)
+
+#define LCDC_LAYER0_OFFSETX_REG_SET_FIELD(_field, _var, _val)\
+	((_var)) =	\
+	((_var) & ~(LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Msk)) |	\
+	(((_val) << LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Pos) &	\
+	LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Msk)
+
+#if (DT_INST_PROP(0, pixel_format) == PANEL_PIXEL_FORMAT_RGB_565)
```

Comment:
Instead of using this approach, we should use the `DISPLAY_BITS_PER_PIXEL` macro to determine the pixel size